stdlib/src/builtin/string.mojo

```diff
@@ -1405,36 +1441,172 @@ struct String(
             substr._strref_dangerous(), start=start
         )
 
-    fn split(self, delimiter: String) raises -> List[String]:
-        """Split the string by a delimiter.
+    fn isspace(self) -> Bool:
+        """Determines whether the given String is a python
+        whitespace String. This corresponds to Python's
+        [universal separators](
+            https://docs.python.org/3/library/stdtypes.html#str.splitlines)
+        `" \\t\\n\\r\\f\\v\\x1c\\x1e\\x85\\u2028\\u2029"`.
+
+        Returns:
+            True if the String is one of the whitespace characters
+                listed above, otherwise False."""
+        return isspace_python(self)
+
+    fn split(self, owned separator: String, maxsplit: Int = -1) -> List[String]:
```

Comment:
nit: use `sep` instead of `separator` for python compatilibitly. Also, why does this have to be `owned`?
```suggestion
    fn split(self, sep: String, maxsplit: Int = -1) -> List[String]:
```